File: Task_1/task_1.kt
fun getRiverCrossingResult(): Int {
    print("Введите общее количество валунов (stones): ")
    val stones = readln().toInt() // Читаем количество валунов

    print("Введите количество птиц (N): ")
    val n = readln().toInt()     // Читаем количество птиц

    val birds = IntArray(n) // Создаем массив для хранения дальности полета птиц
    print("Введите дальность полета для каждой из $n птиц через пробел: ")
    val birdDistances = readln().split(" ").map { it.toInt() } // Читаем дальности полета и преобразуем в массив чисел
    for (i in 0..<n) {
        birds[i] = birdDistances[i] // Заполняем массив birds
    }

    val visitedStones = BooleanArray(stones) // Создаем массив для отслеживания посещенных валунов.
                                             // Изначально все значения false
    // Моделируем перелеты птиц
    for (range in birds) {
        for (i in range - 1..<stones step range) {
            visitedStones[i] = true // Помечаем валун как посещенный
        }
    }

    var unvisitedCount = 0
    for (visited in visitedStones) {
        if (!visited) {       // Если валун не посещен
            unvisitedCount++ // Увеличиваем счетчик
        }
    }

    return unvisitedCount // Возвращаем количество непосещенных валунов
}

fun main() {
    val result = getRiverCrossingResult() // Вызываем функцию
    println("Количество валунов, на которых не появится ни одной птицы: $result") // Выводим результат
}

File: Task_1/task_1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

// Функция для ввода данных и вычислений
func getRiverCrossingResult() int {
	var stones, n int

	fmt.Print("Введите общее количество валунов (stones): ")
	fmt.Scanln(&stones)

	fmt.Print("Введите количество птиц (N): ")
	fmt.Scanln(&n)

	birds := make([]int, n)
	fmt.Printf("Введите дальность полета для каждой из %d птиц через пробел: ", n)

	reader := bufio.NewReader(os.Stdin)
	birdDistancesStr, _ := reader.ReadString('\n')
	birdDistances := strings.Split(strings.TrimSpace(birdDistancesStr), " ")

	for i := 0; i < n; i++ {
		if i < len(birdDistances) {
			num, err := strconv.Atoi(birdDistances[i])
			if err == nil {
				birds[i] = num
			} else {
				birds[i] = 0
			}
		} else {
			birds[i] = 0
		}
	}

	visitedStones := make([]bool, stones) // false означает, что валун не посещен

	for _, rangeVal := range birds {
		for i := rangeVal - 1; i < stones; i += rangeVal {
			visitedStones[i] = true
		}
	}

	unvisitedCount := 0
	for _, visited := range visitedStones {
		if !visited {
			unvisitedCount++
		}
	}

	return unvisitedCount
}

func main() {
	result := getRiverCrossingResult()
	fmt.Println("Количество валунов, на которых не появится ни одной птицы:", result)
}
